src/aui/auibook.cpp

```diff
@@ -3446,6 +3446,127 @@ bool wxAuiNotebook::InsertPage(size_t index, wxWindow *page,
     }
 }
 
+class wxAuiLayoutObject
+{
+public:
+    wxAuiLayoutObject(wxSize &s, unsigned short l, unsigned short dir, unsigned short r, unsigned short pos)
```

Comment:
Maybe a comment would explain this, but do we really have to use a non-const reference here? This makes the code difficult to understand, could we just return `wxSize` from `Merge()` instead?

Also, using `layer` instead of `l`, `row` instead of `r` etc would make this ctor signature much more understandable. Alternatively, after looking at the code below, it seems like it could just take a single `wxAuiPaneInfo` parameter instead.

Reply:
You're right. A `const` argument is all we need here as only the copied member is changed. Returning a `wxSize` from `MergeLayout` would work, too. But I am merging the values inside the object (also multiple times, if necessary). Otherwise I would need to pass the merged `wxSize` value back to each of the `MergeLayout` calls.